projects/gateway2/utils/delegation.go

```diff
@@ -0,0 +1,122 @@
+package utils
+
+import (
+	"reflect"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+	gwv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	"github.com/solo-io/gloo/projects/gateway2/wellknown"
+)
+
+// ChildRouteCanAttachToParentRef returns a boolean indicating whether the given delegatee/child
+// route can attach to a parent referenced by its NamespacedName.
+//
+// A delegatee route can attach to a parent if either of the following conditions are true:
+//   - the child does not specify ParentRefs (implicit attachment)
+//   - the child has an HTTPRoute ParentReference that matches parentRef
+func ChildRouteCanAttachToParentRef(
+	route *gwv1.HTTPRoute,
+	parentRef types.NamespacedName,
+) bool {
+	if route == nil {
+		return false
+	}
+
+	childParentRefs := route.Spec.ParentRefs
+
+	// no explicit parentRefs, so any parent is allowed
+	if len(childParentRefs) == 0 {
+		return true
+	}
+
+	// validate that the child's parentRefs contains the specified parentRef
+	for _, ref := range childParentRefs {
+		// default to the child's namespace if not specified
+		refNs := route.Namespace
+		if ref.Namespace != nil {
+			refNs = string(*ref.Namespace)
+		}
+		// check if the ref matches the desired parentRef
+		if ref.Group != nil && *ref.Group == wellknown.GatewayGroup &&
+			ref.Kind != nil && *ref.Kind == wellknown.HTTPRouteKind &&
+			string(ref.Name) == parentRef.Name &&
+			refNs == parentRef.Namespace {
+			return true
+		}
+	}
+	return false
+}
+
+// ShouldChildRouteInheritParentMatcher returns true if the child HTTPRoute should inherit the parent HTTPRoute's matcher
+func ShouldChildRouteInheritParentMatcher(route *gwv1.HTTPRoute) bool {
+	if route == nil {
+		return false
+	}
+	val, ok := route.Annotations[wellknown.InheritMatcherAnnotation]
+	if !ok {
+		return false
+	}
+	switch strings.ToLower(val) {
+	case "true", "yes", "enabled":
```

Comment:
we can use [IsEnvTruthy](https://github.com/solo-io/gloo/blob/a8ab7afe80ddba7c1b2261785b624c0b083eccbc/pkg/utils/envutils/env.go#L10) here

Reply:
It would not be backward compatible as it would return false for `enabled`.